docs/web/sdk-component-reference.mdx

```diff
@@ -0,0 +1,384 @@
+---
+title: Component Reference
+description: Geocortex Web - Reference for custom components in Geocortex Web
+---
+
+import useBaseUrl from "@docusaurus/useBaseUrl";
+import UseCaseCard from "../../src/components/UseCaseCard";
+import UseCaseContainer from "../../src/components/UseCaseContainer";
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+The following is an example of a custom component that has been created using the Geocortex Web SDK. In general, Geocortex Web components are functional React components extended with a few new behaviors and patterns.
+
+```tsx
+import React, { useState } from "react";
+import {
+    LayoutElement,
+    LayoutElementProperties,
+} from "@geocortex/web/components";
+import List from "@geocortex/web/ui/list";
+import ListItem from "@geocortex/web/ui/list-item";
+import TitleBar from "@geocortex/web/ui/title-bar";
+import Button from "@geocortex/web/ui/button";
+import { useWatchAndRerender } from "@geocortex/web/ui/hooks";
+import DialogActions from "@geocortex/web/ui/dialog-actions";
+import ExampleComponentModel from "./ExampleComponentModel";
+
+export default function ExampleComponent(
+    props: LayoutElementProperties<ExampleComponentModel>
+) {
+    const { model } = props;
+    const [hidden, setHidden] = useState(false);
+
+    useWatchAndRerender(model, "items");
+
+    var x = model.items.map((item, idx) => (
+        <ListItem key={idx} title={item}></ListItem>
+    ));
+
+    return (
+        <LayoutElement {...props}>
+            <List>
+                <DialogActions key="dialog-actions">
+                    <TitleBar text="Some Title"></TitleBar>
+                    {hidden && (
+                        <Button onClick={() => setHidden(false)}>
+                            Show Component
+                        </Button>
+                    )}
+                    {!hidden && (
+                        <Button onClick={() => setHidden(true)}>
+                            Hide Component
+                        </Button>
+                    )}
+                </DialogActions>
+                {!hidden &&
+                    model.items
+                        .map((item, idx) => (
+                            <ListItem key={idx}>{item}</ListItem>
+                        ))
+                        .toArray()}
```

Comment:
I'm guessing `map()` is one of the `Collection` methods and is returning another `Collection`?

Reply:
That's correct. `Collection` mostly mirrors the methods on `Array.prototype`, but they return `Collection` instead.

Reply:
Cool thanks for clearing that up.